frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -546,6 +546,7 @@ end
 function ReaderFooter:onPageUpdate(pageno)
     self.pageno = pageno
     self.pages = self.view.document:getPageCount()
+    self.ui.doc_settings:saveSetting("doc_pages", self.pages) -- for Book information
```

Comment:
We dont't need to save the page count every time page turns.
But note that:
```lua
    --- Saves a setting.
    function DocSettings:saveSetting(key, value)
        self.data[key] = value
    end
```

so `saveSetting` just updates a key, it does not save the setting to disk (``DocSettings:flush()`` does that).
What I need is to have, in the sidecar file, the number of pages of the book. This number of pages is fixed for PDF, but it may change for crengine docs, and it does when you change fonts, switch orientation... And I want it updated in the settings data structure when that happen, so it is saved to the sidecar file when one leaves Reader.
Logically, we may find out all the function that may have the number of pages changed, and add ``self.ui.doc_settings:saveSetting("doc_pages", self.view.document:getPageCount())`` ti each of them, but we may miss a few.
But when any of these happen (font, orientation, hyphenation, floating punctuation), `PageUpdate` event is send, and all `onPageUpdate` are called (it's also called when you just turn page, as you said).
And in `readerfoot.lua` , we have:
```lua
function ReaderFooter:onPageUpdate(pageno)
    self.pageno = pageno
    self.pages = self.view.document:getPageCount()
    self:updateFooterPage()
end
```

and it is actually the only ``onPageUpdate`` that is already calling ``self.view.document:getPageCount()``, so, by putting what I put there, I put it just once, and I avoid the cost of doing more calls to  ``document:getPageCount()``.
I could also put it more logically in both readerrolling and readerpaging, or in the single readerview, for the cost of two or one additionnal ``document:getPageCount()``.

(I don't get what fastDigest() does, and how it could help with what I want to do.)